packages/tooltip/src/Tooltip/Tooltip.tsx

```diff
@@ -129,7 +136,11 @@ function Tooltip({
           return {
             onMouseEnter: debounce((e: MouseEvent) => {
               userTriggerHandler('onMouseEnter', e);
-              setOpen(true);
+              // Without this the tooltip sometimes opens without a transition. flushSync prevents this state update from automically batching. Instead updates are made synchronously.
```

Comment:
Is there a way to test this reliably, to make sure we don't cause a regression in the future?